Window.RequestFocus() does not work (invalid memory address or nil pointer dereference)

<!-- Please search for open issues that relate to the same problem before opening a new one. -->

### Describe the bug:
Using this throws an error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x700e14]

goroutine 1 [running, locked to thread]:
github.com/go-gl/glfw/v3.3/glfw.(*Window).Focus.func1(0x4b?)
	/home/krik/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20211024062804-40e447a793be/window.go:607 +0x14
github.com/go-gl/glfw/v3.3/glfw.(*Window).Focus(0xc000412038?)
	/home/krik/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20211024062804-40e447a793be/window.go:607 +0x19
fyne.io/fyne/v2/internal/driver/glfw.(*window).create.func3()
	/home/krik/go/pkg/mod/fyne.io/fyne/v2@v2.1.4/internal/driver/glfw/window.go:1495 +0x3b0
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0xc0000764e0)
	/home/krik/go/pkg/mod/fyne.io/fyne/v2@v2.1.4/internal/driver/glfw/loop.go:127 +0x18c
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0xc000412000?)
	/home/krik/go/pkg/mod/fyne.io/fyne/v2@v2.1.4/internal/driver/glfw/driver.go:85 +0x30
fyne.io/fyne/v2/internal/driver/glfw.(*window).ShowAndRun(0xc000412000)
	/home/krik/go/pkg/mod/fyne.io/fyne/v2@v2.1.4/internal/driver/glfw/window.go:479 +0x2f
main.main()
	/home/krik/tmp/fynetest/main.go:13 +0x98
```

### To Reproduce:
Steps to reproduce the behaviour:
1. Write example code with window.RequestFocus()
2. Compile
3. Run
5. See error

### Example code:
```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Test")
	w.RequestFocus()
	w.SetContent(widget.NewLabel("Test"))
	w.ShowAndRun()
}
```


### Device (please complete the following information):
 - **OS:** Linux
 - **Version:** Arch
 - **Go version:** 1.18
 - **Fyne version:** 2.1.4

--- a/internal/driver/glfw/window_desktop.go
+++ b/internal/driver/glfw/window_desktop.go
@@ -166,7 +166,7 @@ func (w *window) doCenterOnScreen() {
 }
 
 func (w *window) RequestFocus() {
-	if isWayland {
+	if isWayland || w.view() == nil {
 		return
 	}
 

